posteriors/utils.py

```diff
@@ -1012,7 +1010,11 @@ def f_per_sample(params, batch):
         batch = tree_map(lambda x: x.unsqueeze(0), batch)
         return f(params, batch)
 
-    return f_per_sample
+    @wraps(f)
+    def f_per_sample_ensure_no_kwargs(params, batch):
```

Comment:
since you already say in the comment that it requires no kwargs, maybe there is no need to also use have `ensure_no_kwargs` in the name?

Reply:
but I have to call it something 😅

Reply:
Maybe: `f_per_sample_no_kwargs`? (I don't have a strong opinion on this)

Reply:
I think its fine as it is, its fairly descriptive and entirely internal anyway